src/Microsoft.OpenApi/Helpers/DictionaryCloneHelper.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.OpenApi.Helpers
+{
+    /// <summary>
+    /// Helper class for deep cloning dictionaries.
+    /// </summary>
+    public class DictionaryCloneHelper
+    {
+        /// <summary>
+        /// Deep clone key value pairs in a dictionary.
+        /// </summary>
+        /// <typeparam name="T">The type of the key of the dictionary.</typeparam>
+        /// <typeparam name="U">The type of the value of the dictionary.</typeparam>
+        /// <param name="dictionary">The target dictionary to clone.</param>
+        /// <returns>The cloned dictionary.</returns>
+        public static Dictionary<T, U> Clone<T, U>(IDictionary<T, U> dictionary)
+        {
+            var clonedDictionary = new Dictionary<T, U>();
```

Comment:
return null when the value is null early on. This will have multiple benefits:
- no undesired transforming null values into empty 
- reduced arrow head
- you can now init the new dictionary with a size right away to avoid dictionary growth (moving a collection dictionary in memory is "costly")